yarn-project/aztec-nr/aztec/src/state_vars/immutable_singleton.nr

```diff
@@ -27,46 +28,45 @@ impl<Note, N> ImmutableSingleton<Note, N> {
     ) -> Self {
         assert(storage_slot != 0, "Storage slot 0 not allowed. Storage slots must start from 1.");
         ImmutableSingleton {
-            context,
+            context: context.private,
             storage_slot,
             note_interface,
+            compute_initialization_nullifier: compute_singleton_initialization_nullifier,
         }
     }
     // docs:end:new
 
-    unconstrained fn is_initialized(self) -> bool {
-        let nullifier = self.compute_initialisation_nullifier();
-        oracle::notes::is_nullifier_emitted(nullifier)
+    // docs:start:is_initialized
+    unconstrained fn is_initialized(self, owner: Option<Field>) -> bool {
+        let compute_initialization_nullifier = self.compute_initialization_nullifier;
+        let nullifier = compute_initialization_nullifier(self.storage_slot, owner);
+        check_nullifier_exists(nullifier)
     }
+    // docs:end:is_initialized
 
     // docs:start:initialize
-    fn initialize(self, note: &mut Note) {
+    fn initialize(self, note: &mut Note, owner: Option<Field>) {
+        let context = self.context.unwrap_unchecked();
```

Comment:
Maybe we should do regular unwrap? if someone calls this on a public function we probably want the check to fail at compile time

Reply:
Yeah that makes sense!